version/derive/src/lib.rs

```diff
@@ -91,100 +82,64 @@ pub fn version_with_json(attr: TokenStream, item: TokenStream) -> TokenStream {
 ///    _ => Err("Unsupported version.".to_string()),
 /// }.unwrap();
 /// ```
+#[manyhow]
 #[proc_macro]
-pub fn declare_versioned(input: TokenStream) -> TokenStream {
-    let args = parse_macro_input!(input as DeclareVersionedArgs);
-    impl_declare_versioned(&args, true, true).into()
+pub fn declare_versioned(input: TokenStream) -> Result<TokenStream> {
+    let args = syn2::parse2(input)?;
+    Ok(impl_declare_versioned(&args, true, true))
 }
 
 /// See [`declare_versioned`](`declare_versioned()`) for more information.
+#[manyhow]
 #[proc_macro]
-pub fn declare_versioned_with_scale(input: TokenStream) -> TokenStream {
-    let args = parse_macro_input!(input as DeclareVersionedArgs);
-    impl_declare_versioned(&args, true, false).into()
+pub fn declare_versioned_with_scale(input: TokenStream) -> Result<TokenStream> {
+    let args = syn2::parse2(input)?;
+    Ok(impl_declare_versioned(&args, true, false))
 }
 
 /// See [`declare_versioned`](`declare_versioned()`) for more information.
+#[manyhow]
 #[proc_macro]
-pub fn declare_versioned_with_json(input: TokenStream) -> TokenStream {
-    let args = parse_macro_input!(input as DeclareVersionedArgs);
-    impl_declare_versioned(&args, false, true).into()
+pub fn declare_versioned_with_json(input: TokenStream) -> Result<TokenStream> {
+    let args = syn2::parse2(input)?;
+    Ok(impl_declare_versioned(&args, false, true))
 }
 
-fn impl_version(args: Vec<NestedMeta>, item: TokenStream) -> TokenStream2 {
-    let (item, struct_name) = if let Ok(item_struct) = syn::parse::<ItemStruct>(item.clone()) {
-        (quote!(#item_struct), item_struct.ident)
-    } else if let Ok(item_enum) = syn::parse::<ItemEnum>(item) {
-        (quote!(#item_enum), item_enum.ident)
-    } else {
-        abort_call_site!("The attribute should be attached to either struct or enum.");
-    };
-    let args_map: FxHashMap<_, _> = args
-        .into_iter()
-        .filter_map(|meta| {
-            if let NestedMeta::Meta(Meta::NameValue(name_value)) = meta {
-                Some((
-                    name_value
-                        .path
-                        .get_ident()
-                        .expect_or_abort("Expected single identifier for attribute argument key.")
-                        .to_string(),
-                    name_value.lit,
-                ))
-            } else {
-                None
-            }
-        })
-        .collect();
+#[derive(FromMeta)]
+struct VersionArgs {
+    n: u32,
+    versioned: syn2::Ident,
+}
```

Comment:
Just give them the right names. `n` is **very** confusing, especially once it becomes ambiguous.

Reply:
Should I just change the macro interface?

Reply:
Yes please